common/authorization/opa_claim_mapper.go

```diff
@@ -0,0 +1,51 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package authorization
+
+type (
+	opaClaimMapper struct {
+	}
+
+	opaClaimExtensions struct {
+		AuthInfo AuthInfo `json:"authInfo"`
```

Comment:
The TLS fields in here seem like they could get pretty big. Does it really make sense to pass full client cert chains (keeping in mind that they've already been validated here)? If not, should we maybe copy this struct and clear some of the larger and less useful fields?

`TLSSubject.Names` is probably not useful since the common ones are parsed into the fields above it (and of course everyone only uses CommonName)

`TLSConnection.State.PeerCertificates`, `VerifiedChain`, etc. could be quite large and probably were not intended to be encoded as JSON

Do you have an example of what a typical value looks like in JSON for TLS and non-TLS?

Reply:
Good point, I'll have a look into this. So long as we provide the FQDN I guess that should be enough for most cases, maybe we should include the fingerprint too?